### ISSUE TITLE
YAML definition of Groovy POGO with enum field includes two unexpected values

### ISSUE THREAD

**Issue by tcrespog** [micronaut-projects/micronaut-openapi#358]
### Task List

- [x] Steps to reproduce provided
- [x] Example that reproduces the problem uploaded to Github
- [x] Full description of the issue provided (see below)

### Steps to Reproduce

1. Create a Groovy POGO with any enum fields.
2. Generate the OpenAPI docs.
3. Go to the generated YAML definition file: the enum element includes two additional values (`MIN_VALUE` and `MAX_VALUE`).

### Expected Behaviour

The enum element should contain only the values defined for it. For example:
```groovy
class Pet {
    String name
    Type type
}
```
```groovy
enum Type {
    DOG, CAT
}
```
Corresponds to:
```yml
components:
  schemas:
    Pet:
      type: object
      properties:
        name:
          type: string
        type:
          $ref: '#/components/schemas/Type'
    Type:
      type: string
      enum:
      - DOG
      - CAT
```

### Actual Behaviour

The YAML definition for the enum contains two additional values, `MIN_VALUE` and `MAX_VALUE`:
```yml
components:
  schemas:
    Pet:
      type: object
      properties:
        name:
          type: string
        type:
          $ref: '#/components/schemas/Type'
    Type:
      type: string
      enum:
      - DOG
      - CAT
      - MIN_VALUE
      - MAX_VALUE
```

### Environment Information

- **Operating System**: Ubuntu 20.04.
- **Micronaut Version:** 2.1.0
- **JDK Version:** 1.8

### Example Application

- https://github.com/tcrespog/mn2-openapi-enums
- Compile the project in the `master` branch. Go to the API YML.
- The `java` branch is the Java version of the app and works as expected.

**Comment by ilopmar** [comment]
This is an issue in core https://github.com/micronaut-projects/micronaut-core/issues/6760

**Comment by ilopmar** [comment]
Fixed in Micronaut 3.3.0-M1
